### PR TITLE
Update consul version and deploy in prod mode.

### LINKED ISSUE #1: Consul is not retaining configuration between restarts
Consul Docker image by default is runs in `DEV` mode a doesn't save configuration to file. 
```
# By default you'll get an insecure single-node development server that stores
# everything in RAM, exposes a web UI and HTTP endpoints, and bootstraps itself.
# Don't use this configuration for production.
CMD ["agent", "-dev", "-client", "0.0.0.0"]
```

Add the following to the compose files for Consul to override the default so that it saves configuration between restarts.
` command: agent -ui -data-dir /consul/data -client 0.0.0.0 -server -bootstrap`

### LINKED ISSUE #15: Version bump of consul for Geneva
Version bump of consul for Geneva

Cloned from https://github.com/edgexfoundry/edgex-go/issues/1935.

Bump consul version in custom consul container.
Snaps must be similarly updated.
aarch64 docker-compose may require special treatment (doesn't use custom consul container for non-security build)

Version bump is needed to pick up security fixes and reduce technical debt from using out-of-date container versions.

Blackbox regressions must be run against new version.

### PR DESCRIPTION
**Pull request by bnevis-i**
Update consul version and deploy in prod mode.

Fixes #1.  (originally https://github.com/edgexfoundry/edgex-go/issues/1935)
Fixes #15. (Consul version bump)

Bump the consul version to the latest
and override the CMD argument so that it deploys
in production mode--the default for Consul
is to deploy in -dev mode.  In -dev mode,
consul forgets it state between runs.

Signed-off-by: Bryon Nevis <bryon.nevis@intel.com>

### CODE REVIEW

**Comment by bnevis-i** [comment]
Assistance running blackbox tests is needed to ensure that nothing has been broken by this change.

cc: @jim-wang-intel, @lenny-intel, @tsconn23

File: VERSION [hunk 1]
```diff
@@ -1 +1 @@
-1.1.0
\ No newline at end of file
+master
```
Comment (lenny-goodell) [inline VERSION]: Per @lranjbar , just leave as is. CI is already tagging image with master. Version file will eventually be removed.

**Review by lenny-goodell — DISMISSED** [review]

File: VERSION (hunk 1 above)
Comment (bnevis-i) [reply VERSION]: That is true, but "make docker" will still create "1.1.0".  Shouldn't we be consistent for local builds?

**Comment by bnevis-i** [comment]
P.S. I have NOT done any validation on this PR (blackbox tests currently broken).

File: VERSION (hunk 1 above)
Comment (lenny-goodell) [reply VERSION]: That is only for building a local docker image. Really should be set to `dev` in the makefile, but all that is changing as DevOps executes their CI changes.

Comment (bnevis-i) [reply VERSION]: Backed out the change to VERSION.

**Comment by bnevis-i** [comment]
recheck

**Comment by bnevis-i** [comment]
@lranjbar  This CI job continuing to fail.

**Comment by lranjbar** [comment]
This CI build infrastructure for this repo has been fixed. Rebase off master to rebuild.

**Comment by bnevis-i** [comment]
@lranjbar There doesn't appear to be a docker-edgex-consul-verify-pipeline -- rebased commit still fails.

**Comment by bnevis-i** [comment]
@lranjbar  I am going to close and re-open since no feedback on this PR anyway.

**Pull request closed without merging**